New Tweak: Stacked badges

marcustyphoon opened the pull request:
Q: why
A: it was a more fun use of half an hour than trying to instrument an intentionally-delayed domcontentloaded event okay

### Description
<!--
  What is the goal of this pull request?
  How does it achieve that goal?
  Any other context needed to understand the changes?

  Please properly link any issues that this PR aims to resolve:
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

Adds a tweak that stacks the "important internet checkmarks" unless you hover over them.

<img width="571" src="https://user-images.githubusercontent.com/8336245/217405795-daea420f-bf8c-4991-8685-5bf1401b6029.png">

~~Slightly out of date~~ hover animation demo:

<img width="571" src="https://user-images.githubusercontent.com/8336245/217402956-c880804a-a69a-4ecd-a75b-0ba884b8c07d.gif">

### Testing steps
<!--
  What is the intended behaviour of this pull request?
  How exactly can a maintainer reproduce it?

  Please assume your reviewer will load the addon in a temporary profile.
  Feel free to upload a configuration file if the setup is complex.
-->

- Confirm that badges are stacked, can be hovered, animate correctly, and still have popups when hovered.
- Confirm that stacked badges appear correctly in the regular and community header.

marcustyphoon committed:
Create checkmark_stack.js

marcustyphoon committed:
Prevent animation on page load

marcustyphoon committed:
cleanup

marcustyphoon committed:
Merge remote-tracking branch 'upstream/master' into checkmark-stack

marcustyphoon committed:
update dynamic style insertion

marcustyphoon committed:
only apply drop shadow when stacked

makes the hovered appearance identical to not having this script

marcustyphoon committed:
refactor

marcustyphoon commented:
Note that this is now broken by the (very nice! good job!) `tooManyCheckmarks` css used to achieve a lesser but somewhat similar effect by redpop if someone has a ton.

 - [ ] override this

```css
.EIL5u .tDT3c {
    top: 2px;
    left: calc(2% + 95%*var(--checkmarks-index)/var(--checkmarks-total));
    position: absolute;
}
```

marcustyphoon committed:
update js for "badge" rename

marcustyphoon committed:
update filename for "badge" rename

marcustyphoon committed:
exclude blog headers

marcustyphoon committed:
Revert "exclude blog headers"

This reverts commit 8f9071e0d68fc6db9903a885cb4efc83de83e1a0.

marcustyphoon committed:
exclude blog headers (correctly)

apparently you can just. do this. god damn it.

AprilSylph commented on src/scripts/tweaks/badge_stack.js:
```diff
@@ -5,7 +5,7 @@ import { pageModifications } from '../../util/mutations.js';
 const styleElement = buildStyle();
 
-const wrapper = keyToCss('badgesContainer');
+const wrapper = `:is(article ${keyToCss('badgesContainer')})`;
```
...what is this `:is()` doing?

AprilSylph replied:
oh wait, I didn't expand the whole file, I see.

marcustyphoon committed:
fix for margin change

marcustyphoon committed:
use --badges-index to simplify

marcustyphoon committed:
ignore tooManyBadges

I tried a lot of methods to handle the totally different css used here... nothing good so far

marcustyphoon committed:
Revert "ignore tooManyBadges"

This reverts commit 8fbd3654e2dea7a60617ee9f777b86924dbdfa1f.

marcustyphoon committed:
disable tooManyBadges (sort of hacky)

marcustyphoon committed:
don't expand on username hover

marcustyphoon committed:
style consistency

marcustyphoon committed:
exclude blog headers in tooManyBadges removal

marcustyphoon committed:
update stacked badges for (more) dom changes

marcustyphoon committed:
Revert "update stacked badges for (more) dom changes"

This reverts commit 7d1221f73e0c0f1ad89f45fcfa60b795030e8bc6.

marcustyphoon committed:
update stacked badges for (more) dom changes

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
partially update stacked badges for (even more) dom changes

marcustyphoon committed:
fix trail headers, wrapping, z-index, etc

marcustyphoon committed:
Update tweaks.json

marcustyphoon committed:
remove shrinkBadges override

marcustyphoon committed:
revert change I don't know why I made

marcustyphoon committed:
tweak hover target

marcustyphoon committed:
use `transition-behavior: allow-discrete`

marcustyphoon commented:
https://developer.mozilla.org/en-US/docs/Web/CSS/transition-behavior

Hey, look, a new CSS thingy. Firefox nightly only at the moment, but probably coming soon. Useful for that old "scroll to bottom button needs to change `visibility` after a delay to match the scroll to top button" situation, possibly? Can't remember how or if we solved that.

marcustyphoon committed:
preserve useful part of comment

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
use automatic `styleElement`

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
update selector

marcustyphoon committed:
update selector for communities

marcustyphoon committed:
fix communities shadow clip

marcustyphoon commented:
- [ ] do something about multiple rows in new header

marcustyphoon committed:
Merge remote-tracking branch 'upstream/master' into checkmark-stack

marcustyphoon committed:
make sure wrapped excessive badges aren't visible

marcustyphoon committed:
Merge branch 'master' into checkmark-stack

marcustyphoon committed:
Merge remote-tracking branch 'upstream/master' into checkmark-stack

marcustyphoon committed:
Merge remote-tracking branch 'upstream/master' into checkmark-stack

marcustyphoon committed:
Merge remote-tracking branch 'upstream/master' into checkmark-stack

marcustyphoon committed:
Merge branch 'master' into checkmark-stack
